From 6286902c946f15b2b2ab66904a26d6e0e8748802 Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Wed, 17 Nov 2021 10:47:30 +0200
Subject: [PATCH] SerDes: s32-gen1: Fix wrong SerDes mode being set

Commit (id: '7d08f51a45fc47e61ccb2') introduced the following
regression:
In 'serdes_init' function, instead of 'SERDES_MODE_PCIE_PCIE',
the 's32_serdes_set_mode' function is later called with
'SERDES_MODE_PCIE_SGMII0'.

This breaks the PCIE functionality in u-boot.

Upstream-Status: Pending

Issue: ALB-8200
Signed-off-by: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pci/serdes_s32gen1.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/drivers/pci/serdes_s32gen1.c b/drivers/pci/serdes_s32gen1.c
index 57b17f8e63..7ded446a42 100644
--- a/drivers/pci/serdes_s32gen1.c
+++ b/drivers/pci/serdes_s32gen1.c
@@ -326,18 +326,13 @@ bool s32_serdes_init(struct s32_serdes *pcie)
 	 */
 	if (pcie->phy_mode == CRSS || pcie->phy_mode == SRIS) {
 		if (IS_SERDES_PCIE(pcie->devtype) &&
-		    !IS_SERDES_SGMII(pcie->devtype)) {
-			if (s32_serdes_set_mode(pcie->dbi, pcie->id,
-						SERDES_MODE_PCIE_PCIE))
-				return false;
-		} else {
+		    !IS_SERDES_SGMII(pcie->devtype))
+			pcie->ss_mode = SERDES_MODE_PCIE_PCIE;
+		else
 			return false;
-		}
 	} else if (IS_SERDES_PCIE(pcie->devtype) &&
 			!IS_SERDES_SGMII(pcie->devtype)) {
-		if (s32_serdes_set_mode(pcie->dbi, pcie->id,
-					SERDES_MODE_PCIE_PCIE))
-			return false;
+		pcie->ss_mode = SERDES_MODE_PCIE_PCIE;
 	} else if (IS_SERDES_PCIE(pcie->devtype) &&
 		   IS_SERDES_SGMII(pcie->devtype)) {
 		if (pcie->xpcs_mode != SGMII_XPCS0 &&
-- 
2.17.1

